src/test/java/StringManipulationTest.java

```diff
@@ -50,100 +69,194 @@ public void testRemoveNthCharacter1() {
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("SURELY nothing will go wrong if startWord is greater than endWord...");
+
+        try {
+            manipulatedstring.getSubStrings(6, 2);
+            fail("No exception thrown");
+        } catch (IllegalArgumentException e) {
+        }
+
+        try {
+            manipulatedstring.getSubStrings(-1, 2);
+            fail("No exception thrown");
+        } catch (IllegalArgumentException e) {
+        }
+
+        try {
+            manipulatedstring.getSubStrings(0, -3);
+            fail("No exception thrown");
+        } catch (IllegalArgumentException e) {
+        }
     }
+
+    // Tests whether an exception is thrown when endWord is out of bounds
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Last test. Out of bounds?");
+        try {
+            manipulatedstring.getSubStrings(0, 10);
+            fail("Out of bounds exception not thrown");
+        } catch (IllegalArgumentException e) {
+            assert true;
+        }
     }
 
     @Test
-    public void testRestoreString1()
-    {
+    public void testRestoreString1() {
         manipulatedstring.setString("art");
-        int [] array;
-        array=new int[]{1,0,2};
+        int[] array;
+        array = new int[] { 1, 0, 2 };
         String restoreString = manipulatedstring.restoreString(array);
         assertEquals(restoreString, "rat");
     }
 
+    // Tests a longer string
     @Test
-    public void testRestoreString2()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString2() {
+        manipulatedstring.setString("calebrector");
+        int[] array = new int[] {0, 5, 1, 4, 3, 2, 6, 7, 8, 9, 10};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "crabelector");
     }
 
+    // Tests another longer string
     @Test
-    public void testRestoreString3()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString3() {
+        manipulatedstring.setString("clinteAstWOod");
+        int[] array = new int[] {10, 1, 12, 9, 5, 7, 8, 6, 0, 8, 2, 11, 3};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "OldWestAction");
     }
 
+    // Tests whether capitalization is preserved
     @Test
-    public void testRestoreString4()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString4() {
+        manipulatedstring.setString("CAPStest");
+        int[] array = new int[] {4, 5, 6, 7, 0, 1, 2, 3};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "testCAPS");
     }
 
+    // Tests whether exceptions are thrown when an index is negative or out of bounds
     @Test
```

Comment:
See line 160 comment. It would help to split this test case into several smaller test cases.